apps/src/sites/studio/pages/scripts/stage_extras.js

```diff
@@ -4,12 +4,36 @@ import StageExtras from '@cdo/apps/code-studio/components/stageExtras/StageExtra
 
 const script = document.querySelector('script[data-extras]');
 const config = JSON.parse(script.dataset.extras);
+const showProjectWidget = JSON.parse(script.dataset.widgetVisible);
+const projectTypes = JSON.parse(script.dataset.widgetTypes);
 
-ReactDOM.render(
-  <StageExtras
-    stageNumber={config.stageNumber}
-    nextLevelPath={config.nextLevelPath}
-    bonusLevels={config.bonusLevels}
-  />,
-  document.querySelector('#stage-extras')
-);
+// Get project data if showing the project widget.
+if (showProjectWidget) {
+  $.ajax({
+    method: 'GET',
+    url: `/v3/channels`,
+    dataType: 'json'
+  }).done(projectLists => {
+    ReactDOM.render(
+      <StageExtras
+        stageNumber={config.stageNumber}
+        nextLevelPath={config.nextLevelPath}
+        bonusLevels={config.bonusLevels}
+        showProjectWidget={showProjectWidget}
+        projectTypes={projectTypes}
+        projectList={projectLists}
+      />,
+      document.querySelector('#stage-extras')
```

Comment:
I like @Bjvanminnen 's suggestion here since it lets you decouple the network request from this component while also eliminating duplicate code. If you know you're going to need this component + network request in another place you could wrap it now, otherwise I think it's fine to wait until you need it.